ynr/apps/sopn_parsing/helpers/extract_pages.py

```diff
@@ -32,3 +40,194 @@ def extract_pages_for_ballot(ballot):
         raise PDFException(
             f"Failed to extract pages for {ballot.sopn.uploaded_file.path} as a PDFSyntaxError was raised"
         )
+
+
+textract_client = boto3.client("textract", region_name="eu-west-2")
+
+
+def start_detection(ballot):
+    """This is Step 1 of the SOPN parsing process using AWS Textract.
+    https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/textract/client/get_document_analysis.html#
+    """
+
+    file_path = ballot.sopn.uploaded_file.path
+    with open(file_path, "rb") as file:
+        file_bytes = bytearray(file.read())
+    region = "eu-west-2"
+    bucket_name = "public-sopns"
+    s3_client = boto3.client("s3", region_name=region)
+    object_key = "test/test_sopn.pdf"
+
+    response = s3_client.put_object(
+        Bucket=bucket_name,
+        Key=object_key,
+        Body=file_bytes,
+    )
+    print(f"Uploaded bytes to s3://{bucket_name}/{object_key}")
+    response = textract_client.start_document_analysis(
+        DocumentLocation={
+            "S3Object": {
+                "Bucket": bucket_name,
+                "Name": object_key,
+            }
+        },
+        FeatureTypes=["TABLES", "FORMS"],
+        OutputConfig={
+            "S3Bucket": "public-sopns",
+            "S3Prefix": "test",
+        },
+    )
+    if response["JobId"]:
+        job_id = response["JobId"]
+        get_job_results(job_id, ballot)
+    else:
+        print("Job failed to start")
+        raise Exception("Job failed to start")
+
+
+def get_job_results(job_id, ballot):
+    """This is Step 2 of the SOPN parsing process using AWS Textract.
+    https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/textract/client/get_document_analysis.html#
+    """
+
+    textract_result = textract_client.get_document_analysis(JobId=job_id)
+    while textract_result["JobStatus"] not in ["SUCCEEDED", "FAILED"]:
+        sleep(5)
+        textract_result = textract_client.get_document_analysis(JobId=job_id)
```

Comment:
Another gotcha here. I think we'd be in danger of getting throttled on Sopn Day. Textract doesn't seem to like to many `GET`s. The recommended solution is to kick off the job, then let textract notify SNS topic (not sure how to wire that up) which sends to a SQS Queue which whoever is doing the get_document_analysis bit pulls from.

Reply:
Yes, @symroe and I spoke about this. We have wired it up in this way for this initial implementing/testing phase, but will look at performance next.